cpt/test/integration/upload_test.py

```diff
@@ -127,3 +127,15 @@ def test_existing_upload_repo_by_name(self):
             self.assertNotIn("already exist, keeping the current remote and its name", self.output)
             repo = self.api.get_remote_by_name("upload_repo")
             self.assertEquals(repo.url, "https://api.bintray.com/conan/conan-community/conan")
+
+    def test_upload_when_tag_is_false(self):
```

Comment:
Could you add a test where the upload is done? In the `upload_checks_test`  you can see some examples of how to use the `TestClient` and `TestServer`

Reply:
sure!

Reply:
Done! I've added TestServer and ClientServer to simulate an upload based on tag branch rule.